dapps/digital-twin-detail/src/components/sharing/SharingOverview.ts

```diff
@@ -0,0 +1,234 @@
+      /* Hint: Container API invites every shared user generally into the member role. As a result
+       * of this, we can simply resolve all members from the rights and roles, to check with which
+       * users anything was shared. */
+      const contract = runtime.contractLoader.loadContract('DataContract', containerAddress);
+      const roleMap = await runtime.rightsAndRoles.getMembers(contract);
+      const unique = Array.from(new Set([].concat(...Object.values(roleMap))));
+
+      // iterate through all permitted users and apply them to the permitted users mapping
+      unique.forEach((address: string): void => {
+        if (address !== runtime.activeAccount) {
+          if (!permittedUsers[address]) {
+            permittedUsers[address] = {
+              type: profileUtils.getProfileType(runtime, address),
+              name: profileUtils.getUserAliasFromAddress(runtime, address),
+              containers: [],
+            };
+          }
+
+          permittedUsers[address].containers.push(containerAddress);
+        }
+      });
+    }));
+
+    // iterate through all resolved profiles and push them into the datsa object
+    await Promise.all(Object.keys(permittedUsers).map(async (address: string): Promise<void> => {
+      this.data.push({
+        address,
+        containerNames: this.getContainerNames(permittedUsers[address].containers),
+        containers: permittedUsers[address].containers,
+        icon: profileUtils.getProfileTypeIcon(await permittedUsers[address].type),
+        name: await permittedUsers[address].name,
+        type: await permittedUsers[address].type,
+      });
+
+      this.sortHandler();
+    }));
+
+    this.loading = false;
+  }
+
+  /**
+   * Maps a list of container addresses to it's names.
+   */
+  getContainerNames(containers: string[]): string[] {
+    return containers.map((address: string) => {
+      const container: DAppContainer = this.$store.state.twin.containerContracts[address];
+      return container?.description?.name || container.name;
+    });
+  }
+
+  /**
+   * Recognize changed sorting and sort the current data with this data.
+   *
+   * @param      {any}  ctx     The context
+   */
+  sortHandler(ctx?: any): void {
+    // apply latest sort changes to the context
+    if (ctx) {
+      const { sortBy, sortDesc: reverse } = ctx;
+      this.sortBy = sortBy;
+      this.reverse = reverse;
+    }
+
+    this.data.sort((a, b) => {
+      let sortResult = 0;
+      if (a[this.sortBy] > b[this.sortBy]) {
+        sortResult = 1;
+      } else if (a[this.sortBy] < b[this.sortBy]) {
+        sortResult = -1;
+      }
+
+      if (this.reverse) {
+        sortResult *= -1;
+      }
+
+      return sortResult;
+    });
+  }
+
+  /**
+   * Return the contacts data, filtered by the type filter.
+   */
+  getTableData(): Array<SharedUserInterface> {
+    return this.data.filter((item: SharedUserInterface) => {
+      if (this.searchTerm
```

Comment:
maybe switch those two `if` I think the text filter is less performant so we can already sort out before using the more performant method.